resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/KubernetesTestComponents.scala

```diff
@@ -69,4 +76,79 @@ private[spark] class KubernetesTestComponents {
       .set("spark.testing", "false")
       .set(WAIT_FOR_APP_COMPLETION, false)
   }
+
+  def getService[T: ClassTag](
+    serviceName: String,
+    namespace: String,
+    servicePortName: String,
+    servicePath: String = ""): T = synchronized {
+    val kubernetesMaster = s"${defaultClient.getMasterUrl}"
+    val httpClient = getHttpClient(kubernetesClient.asInstanceOf[BaseClient])
+
+    val url = s"${
+      Array[String](
+        s"${kubernetesClient.getMasterUrl}",
+        "api", "v1", "proxy",
+        "namespaces", namespace,
+        "services", serviceName).mkString("/")
+    }" +
+      s":$servicePortName$servicePath"
+    val userHome = System.getProperty("user.home")
+    val kubernetesConf = kubernetesClient.getConfiguration
+    val sslContext = SSLUtils.sslContext(kubernetesConf)
+    val trustManager = SSLUtils.trustManagers(kubernetesConf)(0).asInstanceOf[X509TrustManager]
+    HttpClientUtil.createClient[T](Set(url), 5, sslContext.getSocketFactory, trustManager)
+  }
+
+  def getHttpClient(client: BaseClient): OkHttpClient = {
+    val field = classOf[BaseClient].getDeclaredField("httpClient")
+    try {
+      field.setAccessible(true)
+      field.get(client).asInstanceOf[OkHttpClient]
+    } finally {
+      field.setAccessible(false)
+    }
+  }
+}
+
+object TestBackend extends Enumeration {
+  val SingleNode, MultiNode = Value
+}
+
+object KubernetesClient {
+  var defaultClient: DefaultKubernetesClient = _
+  var testBackend: TestBackend.Value = _
```

Comment:
That's a good idea. I'm tempted to punt it for later though, since right now there's still quite a bit of `Minikube` being used directly in a lot of places. Ideally, once we move all of it out and clearly define the boundaries of backends, we can abstract out all the cluster specific operations into `IntegrationTestBackend` abstract parent, or trait. WDYT?